packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -92,6 +92,96 @@ export interface StageOptions extends StageProps {
   readonly placement?: StagePlacement;
 }
 
+/**
+ * Pipeline types.
+ */
+export enum PipelineType {
+  /**
+   * V1 type
+   */
+  V1 = 'V1',
+  /**
+   * V2 type
+   */
+  V2 = 'V2',
+}
+
+/**
+ * Properties of pipeline-Level variable.
+ */
+export interface VariableProps {
+  /**
+   * The name of a pipeline-level variable.
+   */
+  readonly variableName: string;
+
+  /**
+   * The description of a pipeline-level variable. It's used to add additional context
+   * about the variable, and not being used at time when pipeline executes.
+   *
+   * @default - No description.
+   */
+  readonly description?: string;
+
+  /**
+   * The default value of a pipeline-level variable.
+   *
+   * @default - No default value.
+   */
+  readonly defaultValue?: string;
+}
+
+/**
+ * Pipeline-Level variable.
+ */
+export class Variable {
+  private readonly variableName: string;
+  private readonly description?: string;
+  private readonly defaultValue?: string;
+
+  constructor(props: VariableProps) {
+    this.variableName = props.variableName;
+    this.description = props.description;
+    this.defaultValue = props.defaultValue;
+
+    this.validate();
+  }
+
+  private validate() {
+    validatePipelineVariableName(this.variableName);
+    if (
+      this.defaultValue !== undefined
+      && !Token.isUnresolved(this.defaultValue)
+      && (this.defaultValue.length < 1 || this.defaultValue.length > 1000)
+    ) {
+      throw new Error(`Default value for variable '${this.variableName}' must be between 1 and 1000 characters long, got ${this.defaultValue.length}`);
+    }
+    if (this.description !== undefined && !Token.isUnresolved(this.description) && this.description.length > 200) {
+      throw new Error(`Description for variable '${this.variableName}' must not be greater than 200 characters long, got ${this.description.length}`);
+    }
```

Comment:
Only the validation of `variableName` is defined in a separate file (`packages/aws-cdk-lib/aws-codepipeline/lib/private/validation.ts`) in order to use an existing method for regular expressions.